client_cli/src/main.rs

```diff
@@ -458,6 +461,87 @@ mod domain {
                 .wrap_err("Failed to transfer domain")
         }
     }
+
+    mod metadata {
+        use iroha_client::data_model::{self, domain::DomainId};
+
+        use super::*;
+
+        /// A value wrapper that can be parsed from CLI arguments
+        #[derive(Debug, Clone)]
+        pub struct MetadataValue(data_model::Value);
```

Comment:
There is [overlap](https://github.com/hyperledger/iroha/pull/4163/files#diff-d2de845c1f7b57cd77a77f1edd12c5f40cecfb0b9aa239cd629e07c1abe965f9R60) between 2 PRs, only one structure should stay.

Reply:
And I would say that `Value` wrapper from the other PR is more feature-rich in terms of parsing.

Reply:
Then I'll wait for @Asem-Abdelhady PR to be approved & merged and then integrate his changes into this one.